components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/dao/ApplicationDAO.java

```diff
@@ -66,4 +59,13 @@ void changeLifecycle(String applicationUUID, String lifecycleIdentifier, String
     void updateScreenShotCount(String applicationUUID, int tenantId, int count) throws
             ApplicationManagementDAOException;
 
+    Category addCategory(Category category) throws ApplicationManagementDAOException;
+
+    List<Category> getCategories() throws ApplicationManagementDAOException;
```

Comment:
Don't we need to have a separate DAO for category? I think, it will be better to have seperate DAO, as there is a seperate table for this.